src/aglet/target.nim

```diff
@@ -34,20 +42,61 @@ proc height*(target: Target): int {.inline.} =
   ## Returns the height of the target.
   target.size.y
 
-proc clearColor*(target: Target, color: Rgba32f) {.inline.} =
-  ## Clear the target's color with a solid color.
-  target.use()
-  target.gl.clearColor(color.r, color.g, color.b, color.a)
-
-proc clearDepth*(target: Target, depth: float32) {.inline.} =
-  ## Clear the target's depth buffer with a single value.
-  target.use()
-  target.gl.clearDepth(depth)
-
-proc clearStencil*(target: Target, stencil: int32) {.inline.} =
-  ## Clear the target's stencil buffer with a single value.
+proc scissor(target: Target, scissor: Option[Recti]) =
+  if scissor.isSome():
+    let scissor = scissor.get
+    target.gl.scissor(scissor.x, scissor.y, scissor.width, scissor.height)
+  else:
+    target.gl.scissor(0, 0, target.size.x, target.size.y)
+
+proc defaultClearParams*(): ClearParams =
+  ClearParams(
+    color: rgba(0.0, 0.0, 0.0, 1.0).some(),
+    depth: some(1.0.float32),
+    stencil: int32.none(),
+    scissor: Recti.none()
+  )
```

Comment:
Personally I'd make all clears explicit, ie. all of them should default to `none()`. With your current solution it's not possible to disable clearing of the color and depth buffers. I'd also rename this to `clearParams()`, because then we aren't providing any sort of "sane" defaults.

Remember to add documentation comments to all procedures. Here, I'd write something like:
> Returns an empty `ClearParams`. At least one field must be initialized by using `withColor`, `withDepth`, or `withStencil`, otherwise an assertion is raised in `clear()`.